community/cypher/runtime-util/src/main/scala/org/neo4j/cypher/exceptionHandler.scala

```diff
@@ -101,4 +105,11 @@ object exceptionHandler extends MapToPublicExceptions[CypherException] {
   trait RunSafely {
     def apply[T](body: => T)(implicit f: ExceptionHandler = ExceptionHandler.default): T
   }
+
+  def mapToCypher(exception: ValuesException): CypherException = {
+    exception match {
+      case e: UnsupportedTemporalUnitException =>
+        exceptionHandler.cypherTypeException(e.getMessage, e.getCause)
```

Comment:
`e.getCause` will be `null`. You never set it. Also, I think you should use `e` instead.